docs/proposals/quic-design.md

```diff
@@ -0,0 +1,140 @@
+It is suggested that cloudHub should support both websocket and QUIC protocol access at the same time. And the edgehub can choose one of the protocols to access to the cloudhub.
+QUIC is a new transport which reduces latency compared to that of TCP. On the surface, QUIC is very similar to TCP+[TLS](http://technet.microsoft.com/en-us/library/cc785811.aspx)+HTTP/2 implemented on [UDP](http://c3lab.poliba.it/images/3/3b/QUIC_SAC15.pdf). Because TCP is implemented in operating system kernels, and middlebox firmware,making significant changes to TCP is next to impossible.However,since QUIC is built on top of UDP,it suffers from no such limitations.
+Key features of QUIC:
+
+1. [Dramatically reduced connection establishment time](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7867726)
+2. Improved congestion control
+3. [Multiplexing without head of line blocking](https://docs.google.com/document/d/1RNHkx_VvKWyWg6Lr8SZ-saqsQx7rFV-ev2jRFUoVD34/mobilebasic?pli=1)
+4. Forward error correction
+5. Connection migiation
+
+<img src="../images/proposals/quic-tcp-udp.png">
+<img src="../images/proposals/quic-connection.png">
+
+## Motivation
+In edge scenarios , network connectivity could be unstable . With TCP + TLS , it becomes an overhead to establish / re-establish connections frequently due to intermittent networks. In such scenarios, QUIC with its zero RTT can help reduce this overhead and re-establish broken connections faster.
+
+## Use cases
+<img src="../images/proposals/quic-design.png">
+
+### Start the websocket server only
+1. User edit controller.yaml  
+```yaml
+  cloudhub:
+  protocol_websocket: true # enable websocket protocol
+  port: 10000 # open port for websocket server
+  protocol_quic: false # enable quic protocol
+  quic_port: 10001 # open prot for quic server
+  max_incomingstreams: # the max incoming stream for quic server
+  address: 0.0.0.0
+  ca: /etc/kubeedge/ca/rootCA.crt
+  cert: /etc/kubeedge/certs/edge.crt
+  key: /etc/kubeedge/certs/edge.key
+  keepalive-interval: 30
+  write-timeout: 30
+  node-limit: 10
+  ```
+
+2. Running the cloudhub, and start the websocket server.
+3. Waiting for client access through websocket protocol.
+
+### Start the quic server only
+1. User edit controller.yaml
+```yaml
+  cloudhub:
+  protocol_websocket: false # enable websocket protocol
+  port: 10000 # open port for websocket server
+  protocol_quic: true # enable quic protocol
+  quic_port: 10001 # open prot for quic server
+  max_incomingstreams: # the max incoming stream for quic server
+  address: 0.0.0.0
+  ca: /etc/kubeedge/ca/rootCA.crt
+  cert: /etc/kubeedge/certs/edge.crt
+  key: /etc/kubeedge/certs/edge.key
+  keepalive-interval: 30
+  write-timeout: 30
+  node-limit: 10
+  ```
+
+2. Running the cloudhub, and start the quic server.
+3. Waiting for client access through quic protocol.
+
+### Start the websocket and quic server at the same time
```

Comment:
Starting both the websocket and quic servers doesn't seem like a feasible option. Let's allow only one option at a time.